elastic/kibana
# [ILM] Add searchable snapshot field

## Summary

Fix https://github.com/elastic/kibana/issues/83505

Adds functionality for configuring searchable snapshots in the policy UI.

### How to test

1. Start ES with a repo path configured, like: `yarn es snapshot -E path.repo=/tmp/es-snaps`
2. Start Kibana (`yarn start`), go to the "Management" in Kibana and open "Index Lifecycle Policies"
3. Go to create a new policy
4. Enable "Cold phase" and you should see searchable snapshots, but disabled due to license
5. Upload an enterprise license to Elasticsearch and refresh until searchable snapshot is usable in cold (i.e., the toggle appears)
6. Configure searchable snapshot in cold - (optional) you should see a prompt about not having any repositories, follow the instructions to create one use the path you used when you started ES.
7. Give the policy a name and save
8. Open the policy you just created and this time open the advanced settings section in the hot phase and enable searchable snapshots
9. There should be a message about subsequent actions being disabled - this has an effect through the entire form of removing certain fields, like the ability to set searchable snapshot further down in cold (verify this by going to the cold section and seeing that it is now disabled with a callout).
10. Save the policy, re-open it and check that everything was stored correctly.

### Notes

* copied functionality from pre-existing "wait for snapshot policy field"

### Screenshots

<details>

<summary>Hot phase</summary>

**New collapsed with "advanced settings" section**

<img width="1159" alt="Screenshot 2020-12-02 at 12 46 18" src="https://user-images.githubusercontent.com/8155004/100868826-748ef300-349c-11eb-9b73-03c5f2fd5553.png">

**Searchable snapshots disabled in advanced settings**

<img width="1050" alt="Screenshot 2020-12-02 at 12 47 50" src="https://user-images.githubusercontent.com/8155004/100868987-aacc7280-349c-11eb-919b-2244c7e79196.png">

**Searchable snapshots enabled in hot phase with no repository created and information callout**

<img width="1150" alt="Screenshot 2020-12-02 at 12 47 16" src="https://user-images.githubusercontent.com/8155004/100868901-8a9cb380-349c-11eb-96b0-ecbecc4d1154.png">

</details>

<details>

<summary>Cold phase</summary>

**disabled due to license**

<img width="1046" alt="Screenshot 2020-12-02 at 12 52 16" src="https://user-images.githubusercontent.com/8155004/100869360-3f36d500-349d-11eb-8b23-c7f0109f8dab.png">


**enabled in cold**

<img width="1066" alt="Screenshot 2020-12-02 at 12 53 06" src="https://user-images.githubusercontent.com/8155004/100869428-54abff00-349d-11eb-9d1a-e598e0bf8cef.png">

</details>

<details>

<summary>Gif</summary>

![SS-field](https://user-images.githubusercontent.com/8155004/101505978-32baec80-3975-11eb-8a5c-cc3afabc69db.gif)

</details>


## Release note

The ILM policy UI now supports configuring searchable snapshot in the cold phase and in the hot phase.

### Checklist

Delete any items that are not applicable to this PR.
